src/jit/lclvars.cpp

```diff
@@ -3507,9 +3509,14 @@ void Compiler::lvaSortByRefCount()
         }
     }
 
+    if (lvaTrackedToVarNum == nullptr)
+    {
+        lvaTrackedToVarNum = static_cast<unsigned*>(compGetMemArray(lclMAX_TRACKED, sizeof(unsigned), CMK_LvaTable));
```

Comment:
Nice. This means we won't even use this space for minopts.

Reply:
I hope so. I haven't actually verified that minopts doesn't attempt to sort variables :smile:

There may be more room for improvement around sorting. Maybe we don't need both `lvaTrackedToVarNum` and `lvaRefSorted`, they're pretty much the same thing.

Reply:
Seems like we should only need to allocate enough space for the tracked variables, not `lclMAX_TRACKED` space. Currently, we compute lvaTrackedCount below, but it seems like we could compute the total number of tentative tracked vars in the loop above, then cap it at `lclMAX_TRACKED`.

Seems like if we're using "ShortLong" bitsets, and dynamic allocation for things like this, we don't even require `lclMAX_TRACKED` to be constant anymore (though I can't think why we would change that).

Reply:
It would be nice to have a function:
```
unsigned lvaTrackedIndexToVarNum(unsigned index)
{
  asssert(index < lvaTrackedCount);
  return lvaTrackedToVarNum[index];
}
```
just to get that assert.

Reply:
> Seems like we should only need to allocate enough space for the tracked variables, not lclMAX_TRACKED space

This is something I hope to look at in the future. It's probably a bit more complicated because we sort more than once and new variables can be added. We'll probably need to over allocate initially to avoid having to discard the existing array and allocate a new one.

There's also the above mentioned issue that `lvaTrackedToVarNum` and `lvaRefSorted` are redundant, one is enough. And `lvaRefSorted` is allocated every time sorting is done.

And then the use of `qsort` is questionable, its reliance on C-style callbacks makes it kind of expensive, especially on Windows with VC++'s `/cf:guard`.

> It would be nice to have a function:

Yes. Actually more than that:
* `LclVarDsc* lvaGetLclVar(unsigned lclNum);`
* `LclVarDsc* lvaGetLclVarByTrackedIndex(unsigned index);`
* and maybe `unsigned lvaGetLclNum(LclVarDsc* varDsc);`

I find it surprising that in 2018 one can still find code like 
```C++
        unsigned varNum = lvaTrackedToVarNum[deadVarIndex];
        varDsc          = lvaTable + varNum;
```
in a reasonably important codebase.

Reply:
> Nice. This means we won't even use this space for minopts.

@CarolEidt This doesn't seem to be the case. It looks like in minopts we sort twice - once from `lvaMarkLocalVars` and once when doing liveness before LSRA. Without minopts a 3rd sorting happens before SSA (for liveness).

Reply:
Interesting. I thought that we didn't track lclVars in minopts, but it's just that we generally don't enregister them because `opts.compFlags & CLFLG_REGVAR` is false. I wonder why this is ... perhaps this is one of those things that would be beneficial to turn off for Tier 0.